Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,144 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `type`
+    public enum IncidentType: String {
+        case Accident = "accident"
+        case Congestion = "congestion"
+        case Construction = "construction"
+        case DisabledVehicle = "disabled_vehicle"
+        case LaneRestriction = "lane_restriction"
+        case MassTransit = "mass_transit"
+        case Miscellaneous = "miscellaneous"
+        case OtherNews = "other_news"
+        case PlannedEvent = "planned_event"
+        case RoadClosure = "road_closure"
+        case RoadHazard = "road_hazard"
+        case Weather = "weather"
+        
+        case Unknown = "<<unknown type>>"
```

Comment:
Perhaps it would be more correct to represent an unknown kind as a `nil` value for the `kind` property. It would be up to the encoding and decoding code to fall back to this `<<unknown type>>` string, however.

Reply:
In fact, that even simplified the Coding part! Updated code doc to explain `nil` value.